src/components/Articles.jsx

```diff
@@ -11,17 +11,29 @@ const Articles = () => {
 
     const [isLoading, setIsLoading] = useState(true)
     const[articles, setArticles] = useState('')
+    const[topics, setTopics] = useState('')
+    const [searchTopic, setSearchTopic] = useState('')
+    const [isTopicsLoading, setIsTopicsLoading] = useState(true)
 
 
     useEffect(() => {
-        getArticles().then((data) => {
+        getArticles(searchTopic).then((data) => {
             setArticles(data)
             setIsLoading(false)
-            
         })
        .catch((error) => {
         console.log("Error getting articles: ", error);
        })
+    }, [searchTopic]);
+
+    useEffect(() => {
+        getTopics().then((data) => {
+            setTopics(data)
+            setIsTopicsLoading(false)  
+        })
+       .catch((error) => {
+        console.log("Error getting topics: ", error);
+       })
```

Comment:
The logic works great and looks good! However, this articles component has now become quite cluttered! Also there are 2 useEffects here which ideally there should be one for efficiency. I would definitely recommend putting the topics display in its own component and then just rendering that as a child of articles. Will get rid of the clutter and make it more efficient 😁 This isn't urgent so please feel free to fix it after your other tickets if you'd like 🙂